system/security/shared/basesecurity.hpp

```diff
@@ -184,7 +184,7 @@ class CBaseSecurityManager : implements ISecManager, public CInterface
 
     IAuthMap * createSettingMap(IPropertyTree * authconfig)
     {
-        throwUnexpected();
+        return nullptr;
```

Comment:
This doesn't feel right. Seems this should be overridden in the concrete security manager implementation(s) that derive from this class.  The purpose of the throw is to alert a security manager developer that there is a method they need to consider for implementation

Reply:
I considered this, and my initial change did override the method in the htpasswd and singleuser managers (LDAP doesn't subclasses, and already returns NULL). I reconsidered because I didn't think it was proper to break every other third party plugin, assuming any exist, because a previously unused interface is suddenly being called to access optional data.

I did discuss with @afishbeck (before committing the change) and he thought this would be acceptable. I can change it if necessary, but do need to know how the known risk of breaking existing code might affect the version into which this change can be included.

Reply:
My take is that if an implementation is required, it should create a compiler error if not implementated. That way it would never make it to a running system w/o the required implemented methods.

Reply:
I agree with the preference for a compiler error instead of an exception that is less likely to be found. After discussing offline with Russ, the compromise will be to add debug logging to the base class, and override the base class to avoid these messages from the built in managers. Existing implementations will continue to work, but will be nudged to consider implementing the function for themselves.